src/test/CommitUploadProtocolTest.kt

```diff
@@ -0,0 +1,176 @@
+        val lastCommit = Commit(revCommits.first())
+        return lastCommit
+    }
+
+    given("empty repo") {
+        repo.commits = listOf(getLastCommit(git))
+
+        val mockApi = MockApi(mockRepo = repo)
+        val mockConfigurator = MockConfigurator(mockRepos = mutableListOf(repo))
+        RepoHasher(localRepo, mockApi, mockConfigurator).update()
+
+        it("doesn't send added commits") {
+            assertEquals(0, mockApi.receivedAddedCommits.size)
+        }
+
+        it("doesn't send deleted commits") {
+            assertEquals(0, mockApi.receivedDeletedCommits.size)
+        }
+    }
+
+    given("happy path: added one commit") {
+        repo.commits = listOf(getLastCommit(git))
+        val mockApi = MockApi(mockRepo = repo)
+        val mockConfigurator = MockConfigurator(mockRepos = mutableListOf(repo))
+
+        val revCommit = git.commit().setMessage("Second commit.").call()
+        val addedCommit = Commit(revCommit)
+        RepoHasher(localRepo, mockApi, mockConfigurator).update()
+
+        it("doesn't send deleted commits") {
+            assertEquals(0, mockApi.receivedDeletedCommits.size)
+        }
+
+        it("posts one commit as added") {
+            assertEquals(1, mockApi.receivedAddedCommits.size)
+        }
+
+        it("should be that the posted commit is added one") {
+            assertEquals(addedCommit, mockApi.receivedAddedCommits.last())
+        }
+    }
+
+    given("happy path: added few new commits") {
+        repo.commits = listOf(getLastCommit(git))
+        val mockApi = MockApi(mockRepo = repo)
+        val mockConfigurator = MockConfigurator(mockRepos = mutableListOf(repo))
+
+        val otherAuthorsNames = listOf("a", "b", "a")
+        val otherAuthorsEmails = listOf("a@a", "b@b", "a@a")
+        for (i in 0..2) {
+            git.commit().setMessage("Create $i.").setAuthor(otherAuthorsNames.get(i), otherAuthorsEmails.get(i)).call()
+        }
+        val authorCommits = mutableListOf<Commit>()
+        for (i in 0..4) {
+            val message = "Created $i by author."
+            val revCommit = git.commit().setMessage(message).call()
+            authorCommits.add(Commit(revCommit))
+        }
+        RepoHasher(localRepo, mockApi, mockConfigurator).update()
+
+        it("posts five commits as added") {
+            assertEquals(5, mockApi.receivedAddedCommits.size)
+        }
+
+        it("doesn't send deleted commits ") {
+            assertEquals(0, mockApi.receivedDeletedCommits.size)
+        }
+
+        it("processes author's commits") {
+            assertEquals(authorCommits.asReversed(), mockApi.receivedAddedCommits)
+        }
+    }
+
+        given("fork event") {
+
+            val forkedRepoPath = "./forked_repo/"
+            val originalRepoPath = "./original_repo/"
+            val forked = Git.cloneRepository()
+                    .setURI("https://github.com/extropic-engine/Result.git")
```

Comment:
Can you please fork them into our own account? So that tests keep on working even if those repos are removed?